src/otx/engine/engine.py

```diff
@@ -904,3 +915,19 @@ def _get_anomaly_model(
         model.optimizer = optimizer
         model.scheduler = scheduler
         return model
+
+    def _patch_packages_xpu(self) -> None:
+        """Patch packages when xpu is available."""
+        if self.task in [OTXTaskType.DETECTION, OTXTaskType.INSTANCE_SEGMENTATION] and self.device.accelerator == "xpu":
+            import numpy as np
+            from mmcv.ops.nms import NMSop
+            from mmcv.ops.roi_align import RoIAlign
+            from mmengine.structures import instance_data
+
+            from otx.algo.detection.utils import monkey_patched_nms, monkey_patched_roi_align
+
+            long_type_tensor = Union[torch.LongTensor, torch.xpu.LongTensor]
+            bool_type_tensor = Union[torch.BoolTensor, torch.xpu.BoolTensor]
+            instance_data.IndexType = Union[str, slice, int, list, long_type_tensor, bool_type_tensor, np.ndarray]
+            NMSop.forward = monkey_patched_nms
+            RoIAlign.forward = monkey_patched_roi_align
```

Comment:
Is there any reason why this function needs to be a function in Engine? 
It could be declared somewhere like utils and Engine could just call it when it needs it, what do you think?

Reply:
Yes, we can

Reply:
[Done](https://github.com/openvinotoolkit/training_extensions/pull/3094/commits/7e303a1a9caedfa247c9c98d47e7956d5fee01d0)